roxctl/central/debug/resync_check.go

```diff
@@ -0,0 +1,180 @@
+			if err != nil {
+				return err
+			}
+			before, after, err := cmd.run()
+			if err != nil {
+				return err
+			}
+
+			return cmd.storeFiles(before, after)
+		}),
+	}
+	flags.AddTimeoutWithDefault(c, diagnosticBundleDownloadTimeout)
+	c.PersistentFlags().StringVar(&outputDir, "output-dir", "resync-check-output", "output directory in which to store bundle")
+	c.PersistentFlags().DurationVar(&waitFor, "wait-for", time.Minute, "how long to wait between before and after alert check")
+
+	return c
+}
+
+type resyncCheckCmd struct {
+	env       environment.Environment
+	conn      *grpc.ClientConn
+	waitFor   time.Duration
+	outputDir string
+}
+
+func commandWithConnection(env environment.Environment, waitFor time.Duration, outputDir string) (*resyncCheckCmd, error) {
+	conn, err := env.GRPCConnection()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not establish gRPC connection to central")
+	}
+
+	return &resyncCheckCmd{
+		env:       env,
+		conn:      conn,
+		waitFor:   waitFor,
+		outputDir: outputDir,
+	}, nil
+}
+
+func (c *resyncCheckCmd) run() ([]*storage.ListAlert, []*storage.ListAlert, error) {
+	c.env.Logger().InfofLn("Running re-sync check")
+
+	ctx, cancel := context.WithTimeout(context.Background(), resyncCheckTimeout)
+	defer cancel()
+
+	svc := v1.NewAlertServiceClient(c.conn)
+
+	alertsBefore, err := c.fetchAlerts(ctx, svc)
+	if err != nil {
+		return nil, nil, err
+	}
+	c.env.Logger().InfofLn("Found %d alerts before reassessing", len(alertsBefore))
+
+	if err := c.reassessPolicies(ctx); err != nil {
+		return nil, nil, err
+	}
+
+	c.env.Logger().InfofLn("Waiting for %s while reassess policies produces alerts", c.waitFor.String())
+
+	// Roxctl needs to wait here to realistically get the delta between the violation state before and after reassessing
+	// policies. If the timer is too short, there's a higher change that there will be no delta, because the Sensors
+	// still haven't fully reprocessed the deployments.
+	waitForC := time.After(c.waitFor)
+	select {
+	case <-waitForC:
+		break
+	case <-ctx.Done():
+		return nil, nil, errors.Errorf("context finished before second request could be made: `wait-for` value might be too high: %s", c.waitFor.String())
+	}
+
+	alertsAfter, err := c.fetchAlerts(ctx, svc)
+	if err != nil {
+		return nil, nil, err
+	}
+	c.env.Logger().InfofLn("Found %d alerts after reassessing", len(alertsAfter))
+
+	c.assessDelta(alertsBefore, alertsAfter)
+
+	return alertsBefore, alertsAfter, nil
+}
+
+func (c *resyncCheckCmd) fetchAlerts(ctx context.Context, svc v1.AlertServiceClient) ([]*storage.ListAlert, error) {
+	response, err := svc.ListAlerts(ctx, &v1.ListAlertsRequest{})
+	if err != nil {
+		return nil, errors.Wrap(err, "could not make ListAlerts request to gRPC server")
+	}
+	return response.GetAlerts(), nil
+}
+
+func (c *resyncCheckCmd) storeFiles(alertsBefore, alertsAfter []*storage.ListAlert) error {
+	if err := os.Mkdir(c.outputDir, 0755); err != nil {
```

Comment:
Let's short-circuit here and return an error if the directory couldn't be created and `os.IsExist(err) == false`
```suggestion
	if err := os.Mkdir(c.outputDir, 0755); err != nil && !os.IsExist(err){
```